config.c

```diff
@@ -990,21 +990,12 @@ NORETURN
 static void die_bad_number(const char *name, const char *value)
 {
 	const char *error_type = (errno == ERANGE) ?
-		N_("out of range") : N_("invalid unit");
-	const char *bad_numeric = N_("bad numeric config value '%s' for '%s': %s");
+		_("out of range") : _("invalid unit");
+	const char *bad_numeric = _("bad numeric config value '%s' for '%s': %s");
 
 	if (!value)
 		value = "";
 
-	if (!strcmp(name, "GIT_TEST_GETTEXT_POISON"))
-		/*
-		 * We explicitly *don't* use _() here since it would
-		 * cause an infinite loop with _() needing to call
-		 * use_gettext_poison(). This is why marked up
-		 * translations with N_() above.
-		 */
-		die(bad_numeric, value, name, error_type);
```

Comment:
It might make sense to move this code block _first_, in its own commit, and to make it more obvious why it needs to be moved, an explanation in the commit message would be nice.

Reply:
updated the commit message and left in the first commit, what do you think?

Reply:
I had another look, and only now realized that this block was not just moved within a function, but it was moved between functions. That's _kind_ of okay because we know that we only handle that environment variable via `git_config_bool_or_int()`.

But the move also implies that the [entire `switch` block](https://github.com/git/git/blob/v2.28.0/config.c#L1011-L1030) is skipped, which means that other calls to `git_config_bool_or_int()` won't report the origin of the config setting any longer, which is probably not so desirable.

I don't know that it makes sense to continue the review here, as I am not a gate keeper and the _real_ review should happen on the Git mailing list, anyway (i.e. _after_ `/submit`). But if I were to review this, I would suggest to look into extending the signature of `die_bad_number()` (and the entire call chain all the way up to `git_config_bool_or_int()`) to include a parameter indicating whether we asked for a numeric value, for a Boolean, or for either. It would blow up the `switch` block substantially, though.

As I said, it's probably a good idea to send this off to the Git mailing list. It would be good to change https://github.com/git/git/pull/841#issue-486022166 into a proper cover letter that contains the motivation for working on these changes, and the considerations that let you settle for the current solution, maybe soliciting suggestions what other approaches should be considered.

Reply:
thanks for the suggestion, adding the alternative to the PR message